src/modules/CannedMessageModule.cpp

```diff
@@ -546,6 +552,24 @@ void CannedMessageModule::drawFrame(OLEDDisplay *display, OLEDDisplayUiState *st
     }
 }
 
+ProcessMessage CannedMessageModule::handleReceived(const meshtastic_MeshPacket &mp)
+{
+    if (mp.decoded.portnum == meshtastic_PortNum_ROUTING_APP) {
+        // look for a request_id
+        if (mp.decoded.request_id != 0) {
+            UIFrameEvent e = {false, true};
+            e.frameChanged = true;
+            this->runState = CANNED_MESSAGE_RUN_STATE_ACK_RECEIVED;
+            this->incoming = mp.decoded.request_id;
```

Comment:
This is not actually the node number but the message id. We need to dig into the outgoing message store if we find the 'dest' field in it. Right now it always displays 'unknown'

Reply:
About this: we could for now just display "Last message delivered" or "Last delivery failed" until we have a way of finding the `dest` field?